### ISSUE TITLE
Lots of issues with setting run_time=None for infinite simulation

### ISSUE THREAD

**Issue by neworderofjamie** [SpiNNakerManchester/sPyNNaker#74]
This is really useful for roboticsish stuff:

1) vertex.is_set_to_record_spikes check in spinnaker.py - this method doesn't seem to work
2) command_sender.py tries to do something weird about negative times
3) AbstractDataSpecableVertex._write_basic_setup_info doesn't handle None values (presumably the C-code needs some sort of run forever concept too)

**Comment by alan-stokes** [comment]
Hi Jamie. 

My personal opinion on this is that none is a invalid param there. Id rather perfer -1, but i note your concern and will bring it up with Rowley tomorrow morning. As maybe one of us can think up a better solution, or fix the issues in the classes. 

I note that infinite recording is a invalid state, as you'll never have enough sdram to record infitie things..........least until Sergio gets buffered out completed.

**Comment by rowleya** [comment]
We had this discussion some time ago, and agreed that None in the python should be OK (admittedly this was on the SVN version of the toolchain).

The C code should still be able to cope with this, although I can’t remember if it can; the python should set the run time to 0xFFFFFFFF and then the C code should just not terminate if it encounters this value (and in fact it will run for a long time in any case even if this check is not done).

Andrew :)

**Comment by neworderofjamie** [comment]
-1 is also pretty horrible - Infinity would be better i.e. in python
float("inf").
I agree entirely that recording an infinite simulation to SDRAM is invalid,
but I wasn't doing that. I was using live recording AND, unless I was being
really dumb, the is_set_to_record_spikes method it's using to test doesn't
actually exist anywhere...

**Comment by AlexRast** [comment]
I see some long-term problems with allowing truly infinite simulation:

1) How then does the system ensure a "graceful exit"? Remember that pressing the 
reset button is not always going to be an option as you may not have a local board.

2) In a large simulation, operating over a considerable extent of a cabinet 
machine, this would cause problems in job allocation. I don't see how you could 
"permanently" allocate a part of the system for a simulation. We all know the 
simulation will terminate eventually, but in a situation where when that occurs 
can't be predicted, scheduling jobs would be phenomenally different - and a 
large, long simulation could end up hogging resources.

So my feeling is that we might allow very long simulations but not 
infinite-length ones. People need to decide what a reasonable run-time limit for 
their sinulation is.

**Comment by alan-stokes** [comment]
So as i was sorting out getting Jamies push bot pullr equest done. I fixed this by adding an extra param to the system region called infinite runs. This is checked by al models and allows infinite runs. it seemed the code was expecting a max_int to be infinite run, but this would cause issues if end users decided on that runtime acdientally. 

now the changes got reviewed by jamei and were merged this afternoon. So i'm closing this issue as its problem is solved. the "long term problems" i think we can face when it comes about.
